src/compat/crypto.rs

```diff
@@ -4,73 +4,104 @@
 //
 // Copyright (c) DUSK NETWORK. All rights reserved.
 
+use core::mem::size_of;
+
 use dusk_bls12_381::BlsScalar;
 use dusk_bytes::Serializable;
-use phoenix_core::{Note, PublicKey, ViewKey};
+use phoenix_core::{
+    transaction::{ArchivedTreeLeaf, TreeLeaf},
+    Note, PublicKey,
+};
 
-use alloc::vec::Vec;
+use alloc::{string::ToString, vec::Vec};
 
+use crate::alloc::borrow::ToOwned;
 use crate::{
     key::{self},
     types::{self},
     utils::{self},
     MAX_KEY, MAX_LEN,
 };
 
+const TREE_LEAF_SIZE: usize = size_of::<ArchivedTreeLeaf>();
+
 /// Returns true or false if the note is owned by the index
 /// if its true then nullifier of that note if sent with it
 #[no_mangle]
 pub fn check_note_ownership(args: i32, len: i32) -> i64 {
-    // we just use BalanceArgs again as we don't want to add more cluter types
-    // when the data you want is the same
-    let types::CheckNoteOwnershipArgs { note, seed } =
-        match utils::take_args(args, len) {
-            Some(a) => a,
-            None => return utils::fail(),
-        };
+    let args = utils::take_args_raw(args, len);
```

Comment:
Why don't use `from_raw_parts` directly? The overhead to have an additional function call is not worth, there really nothing more than that. If you need the cast, you can either have `.into()` where supported or `as` directly in the `from_raw_parts` call's arguments.

Plus, you're wrongly discard an `unsafe` keyword where is absolutely necessary: here I might be tempted to believe you actually did a safe interface to an `unsafe` call, but it's not the case: you're not performing any checks or doing anything that would imply such thing.